File: src/trocado/features/identity/infrastructure/http/mappers/requests/sign_up_request_mapper.py
from __future__ import annotations

from trocado.features.identity.application.data.sign_up_data import SignUpData
from trocado.features.identity.infrastructure.http.requests.sign_up_request import SignUpRequest


class SignUpRequestMapper:
    @staticmethod
    def to_data(request: SignUpRequest) -> SignUpData:
        return SignUpData(name=request.name, email=request.email, password=request.password)

File: src/main/kotlin/com/bed/cordato/features/identity/infrastructure/http/mappers/SignUpRequestMapper.kt
package com.bed.cordato.features.identity.infrastructure.http.mappers

import com.bed.cordato.features.identity.application.commands.SignUpCommand

import com.bed.cordato.features.identity.infrastructure.http.requests.SignUpRequest

/**
 * Turns the HTTP [SignUpRequest] into the application's [SignUpCommand] — a straight field copy,
 * as an `internal` extension so the call site reads `request.toCommand()`. It carries the raw
 * strings across unchanged: validation is the use case's behavior, never this mapper's.
 */
internal fun SignUpRequest.toCommand(): SignUpCommand = SignUpCommand(
    name = name,
    email = email,
    password = password,
)
